clouddriver-appengine/src/main/groovy/com/netflix/spinnaker/clouddriver/appengine/deploy/validators/DeployAppengineDescriptionValidator.groovy

```diff
@@ -39,20 +39,22 @@ class DeployAppengineDescriptionValidator extends DescriptionValidator<DeployApp
       return
     }
 
-    if (!description.repositoryUrl.startsWith("gs://")) {
+    boolean isContainerDeployment = description.containerImageUrl?.trim()
+
+    if (!isContainerDeployment && !description.repositoryUrl.startsWith("gs://")) {
       if (!helper.validateGitCredentials(description.credentials.gitCredentials,
                                          description.gitCredentialType,
                                          description.credentials.name,
                                          "gitCredentialType")) {
          return
       }
       helper.validateNotEmpty(description.branch, "branch")
+      helper.validateNotEmpty(description.repositoryUrl, "repositoryUrl")
```

Comment:
I think this should live outside this `if`, since the `repositoryUrl` validation won't happen in the case where `repositoryUrl.startsWith("gs://")`.

Also, in the case of `isContainerDeployment`, there should also be a check for `helper.validateNotEmpty(description.containerImageUrl)`.